Makefile

```diff
@@ -1,66 +1,107 @@
+	find $(topbuilddir) -name '*.pyc' -delete
+	find $(topbuilddir) -name '*.py,cover' -delete
 
-deb_dist/$(MODNAME)-$(MODVERSION)/debian/pydist-overrides: debian/pydist-overrides deb_dist
-	ln -s $< $@
+test:
+	tox -c $(topdir)/tox.ini
 
-deb_dist/$(MODNAME)-$(MODVERSION)/debian/py3dist-overrides: debian/py3dist-overrides deb_dist
-	ln -s $< $@
+testenv:
+	tox -r -c $(topdir)/tox.ini --notest
 
-deb: deb_dist/$(MODNAME)-$(MODVERSION)/debian/pydist-overrides deb_dist/$(MODNAME)-$(MODVERSION)/debian/py3dist-overrides
-	cd deb_dist/$(MODNAME)-$(MODVERSION)/; DEB_BUILD_OPTIONS=nocheck debuild -uc -us
-	cp deb_dist/*.deb dist
+$(topbuilddir)/dist:
+	mkdir -p $@
 
-clean:
-	$(PYTHON) setup.py clean || true
-	rm -rf build/ MANIFEST
-	rm -rf dist/
-	rm -rf deb_dist rpm
-	rm -rf .tox
-	rm -rf *.egg-info
-	find . -name '*.pyc' -delete
-	find . -name '*.py,cover' -delete
+deb_dist: $(topbuilddir)/dist/$(MODNAME)-$(VERSION).tar.gz
+	$(PY2DSC) --with-python2=true --with-python3=true $(topbuilddir)/dist/$(MODNAME)-$(VERSION).tar.gz
 
-test:
-	tox -e py27-test-coverage -e py3-test-coverage
+$(DEBIANDIR)/%: $(topdir)/debian/% deb_dist
+	cp -r $< $@
 
-test-nocover:
-	tox -e py27
+dsc: deb_dist $(DEBIANOVERRIDES)
+	cp $(topbuilddir)/deb_dist/$(DEBNAME)_$(VERSION)-1.dsc $(topbuilddir)/dist
+
+deb: source deb_dist $(DEBIANOVERRIDES)
+	cd $(DEBIANDIR)/..;debuild -uc -us
+	cp $(topbuilddir)/deb_dist/python*$(DEBNAME)_$(VERSION)-1*.deb $(topbuilddir)/dist
+
+# START OF RPM SPEC RULES
+# If you have your own rpm spec file to use you'll need to disable these rules
+$(topdir)/rpm/$(MODNAME).spec: rpm_spec
+
+rpm_spec: $(topdir)/setup.py
+	$(PYTHON3) $(topdir)/setup.py bdist_rpm --spec-only --dist-dir=$(topdir)/rpm
+# END OF RPM SPEC RULES
+
+$(RPMBUILDDIRS):
+	mkdir -p $@
+
+$(topbuilddir)/build/rpm/SPECS/$(MODNAME).spec: $(topdir)/rpm/$(MODNAME).spec $(topbuilddir)/build/rpm/SPECS
+	cp $< $@
+
+$(topbuilddir)/build/rpm/SOURCES/$(MODNAME)-$(VERSION).tar.gz: $(topbuilddir)/dist/$(MODNAME)-$(VERSION).tar.gz $(topbuilddir)/build/rpm/SOURCES
+	cp $< $@
+
+rpm: $(topbuilddir)/build/rpm/SPECS/$(MODNAME).spec $(topbuilddir)/build/rpm/SOURCES/$(MODNAME)-$(VERSION).tar.gz $(RPMBUILDDIRS)
+	rpmbuild -ba --define '_topdir $(topbuilddir)/build/rpm' --clean $<
+	cp $(topbuilddir)/build/rpm/RPMS/*/*.rpm $(topbuilddir)/dist
+
+wheel:
+	$(PYTHON2) $(topdir)/setup.py bdist_wheel --dist-dir=$(topbuilddir)/dist
+	$(PYTHON3) $(topdir)/setup.py bdist_wheel --dist-dir=$(topbuilddir)/dist
 
-nosetest:
-	tox -e py27-nose-coverage -e py3-nose-coverage
+egg:
+	$(PYTHON2) $(topdir)/setup.py bdist_egg --dist-dir=$(topbuilddir)/dist
+	$(PYTHON3) $(topdir)/setup.py bdist_egg --dist-dir=$(topbuilddir)/dist
 
-nosetest-nocover:
-	tox -e py27-nose -e py3-nose
+pex:
+	tox -c $(topdir)/tox.ini -e package
 
-.PHONY: test test-nocover nosetest nosetest-nocover clean deb install source all
+.PHONY: test test2 test3 clean install source deb dsc rpm rpm_spec wheel egg pex testenv all
```

Comment:
There is no `test2` or `test3` target. See also earlier comments about pex.

Reply:
Oops, first attempt at that comment got eaten by the Internet going away; apparently that's not how you edit a review!